src/support.js

```diff
@@ -661,14 +661,32 @@ var functionPointers = new Array({{{ RESERVED_FUNCTION_POINTERS }}});
 
 #if WASM
 // Wraps a JS function as a wasm function with a given signature.
-// In the future, we may get a WebAssembly.Function constructor. Until then,
-// we create a wasm module that takes the JS function as an import with a given
-// signature, and re-exports that as a wasm function.
 function convertJsFunctionToWasm(func, sig) {
 #if WASM2JS
   return func;
 #else // WASM2JS
 
+  // If the type reflection proposal is available, use the new
+  // "WebAssembly.Function" constructor.
+  // Otherwise, construct a minimal wasm module importing the JS function and
```

Comment:
Can this be made into a general polyfill?

Reply:
At which level are you imagining this polyfill? Do you suggest implementing a 'WebAssembly.Function' function if it's not defined? That would mostly work, but the returned object would still be of different type I think.

Reply:
Yeah, that's what I thought. I'm not sure specific returned type matters too much (especially in this codebase), but if it does, we can always do `setPrototypeOf` on the returned function.

Reply:
What would be the benefit of doing this? Less code in the `convertJsFunctionToWasm` function?

Reply:
Yeah, but also I thought it might be useful as a generic polyfill for other users that Emscripten just happens to use.